pkg/controller/containerjfr/containerjfr_controller.go

```diff
@@ -277,6 +277,36 @@ func (r *ReconcileContainerJFR) Reconcile(request reconcile.Request) (reconcile.
 		return reconcile.Result{}, err
 	}
 
+	// Check that secrets mounted in /truststore conincide with CRD
+	err = r.Client.Get(context.Background(), types.NamespacedName{Name: deployment.Name, Namespace: deployment.Namespace}, deployment)
+	if err == nil {
+		deploymentSecrets := deployment.Spec.Template.Spec.Containers[0].VolumeMounts
+		secrets := instance.Spec.Secrets
+		for i := 0; i < len(secrets); i++ {
```

Comment:
Typically in Go, you'll see for-each loops of the form `for _, secret := range secrets`. I'd recommend that syntax instead. Same thing for other loops in this PR.